src/pages/loginPage.js

```diff
@@ -1,13 +1,48 @@
-import React from 'react';
+import React, { useRef } from 'react';
 import { useNavigate } from 'react-router-dom';
 
+localStorage.setItem("registeredUsers",JSON.stringify([
```

Comment:
same here, why are you setting registeredUsers every time your loginPage is opened